### ISSUE TITLE
Disabled nullability and MapPropertyAttribute.Use results in invalid generated code

### ISSUE THREAD

**Issue by InspiringCode** [riok/mapperly#1255]
Given the following classes:
```csharp
public class Model
{
    public Material Material { get; set; }
}

public class ModelDto
{
    public string Material { get; set; }
}

public record Material(string Name);
```
When declaring the following mapper:
```csharp
[Mapper]
public partial class DemoMapper
{
    [MapProperty("Material", "Material", Use = nameof(ToMaterial))]
    public partial Model ToModel(ModelDto source);

    [UserMapping(Default = false)]
    public Material ToMaterial(string name) => new Material(name);
}
```

The following invalid code is generated:
```csharp
if (source.Material != null)
{
    target.Material = ToMaterial(source.Material) == null ? throw new System.ArgumentNullException(nameof(ToMaterial(source.Material))) : ToMaterial(source.Material);
}
```

The expression `nameof(ToMaterial(source.Material))` is surely not valid C# syntax and also calling the mapping method twice is not really smart.

**Comment by latonz** [comment]
Thanks for reporting. Are all your classes in a nullable disabled context?

**Comment by InspiringCode** [comment]
Indeed, reference nullability is not enabled for the whole project (because it is a legacy project).

**Comment by InspiringCode** [comment]
@latonz Sorry for the confusion. I was still using version 3.5.0 in my project. I seems to be fixed in 3.5.1, or at least I couldn't reproduce it anymore. Sorry for overlooking this.
